### ISSUE TITLE
Regarding the model: The first encoder is used to calculate the relationship aware embedding of the head entity h hh, while the input of the second encoder BERTt only contains the textual description of entity t.

### ISSUE THREAD

**Issue by whistle9** [intfloat/SimKGC#38]
Hello, I would like to know if the model uses two encoders to obtain the head entity relationship perception embedding and tail entity embedding, and finally performs normalization processing calculation. Can this model be implemented with one encoder？

**Comment by intfloat** [comment]
Yes, we use two separate encoders as shown in `https://github.com/intfloat/SimKGC/blob/97cc43e488f19ca5b0f6fbf60ffefd2ee56c0693/models.py#L43-L44`

You can also implement with one shared encoder, but I did not test its results.

**Comment by whistle9** [comment]
Thank you for your reply. Also, I would like to know if the way you proposed in your paper to add inverse triples to each triplet is to add all triples before negative sampling? Does this have any impact on subsequent negative sampling？

**Comment by whistle9** [comment]
> 是的，我们使用两个独立的编码器，如`https://github.com/intfloat/SimKGC/blob/97cc43e488f19ca5b0f6fbf60ffefd2ee56c0693/models.py#L43-L44`
> 
> 您也可以使用一个共享编码器实现，但我没有测试其结果。
This part of the code initializes two BERT model instances, hr_bert and tail_bert, in the custom BERT model. hr_bert is loaded from the pre trained model, while tail_bert is a deep copy of hr_bert. But what is mentioned in the paper is that the first encoder is used to calculate the relationship aware embedding of the head entity h, and the second encoder is used to calculate the L2 normalized embedding et of the tail entity t. So what is this copy？
